csrank/objectranking/lambdamart.py

```diff
@@ -0,0 +1,376 @@
+import logging, math
+from collections import deque
+from multiprocessing import Pool
+from itertools import chain
+
+import numpy as np
+from sklearn.tree import DecisionTreeRegressor
+
+from csrank.learner import Learner
+from csrank.metrics import point_dcg, dcg, ndcg
+from csrank.objectranking.object_ranker import ObjectRanker
+
+class LambdaMART(ObjectRanker,Learner):
+    def __init__(self, n_objects=None, n_object_features=None, number_of_trees=5, learning_rate=1e-3,
+                 min_samples_split=2, max_depth=50, min_samples_leaf=1, max_leaf_nodes=None, num_process = None,
+                 criterion="mse", splitter="best", min_weight_fraction_leaf=None, max_features=None, random_state=None, 
```

Comment:
`min_weight_fraction_leaf` is equal to `None` by default, but leaving it `None` does not work:
```
~/anaconda3/envs/csrank/lib/python3.6/site-packages/sklearn/tree/tree.py in fit(self, X, y, sample_weight, check_input, X_idx_sorted)
    249             raise ValueError("Number of labels=%d does not match "
    250                              "number of samples=%d" % (len(y), n_samples))
--> 251         if not 0 <= self.min_weight_fraction_leaf <= 0.5:
    252             raise ValueError("min_weight_fraction_leaf must in [0, 0.5]")
    253         if max_depth <= 0:

TypeError: '<=' not supported between instances of 'int' and 'NoneType'
```

Either, prespecify a sensible default value, or raise an exception also for `None` (e.g. `ValueError`). Usually users expect that keyword parameters with default values work out of the box though. So I would go for the default value.